corehq/apps/userreports/transforms/specs.py

```diff
@@ -128,3 +130,16 @@ def transform_function(values):
             return delimiter.join(translated_values_list)
 
         return transform_function
+
+
+class MarkDownTransform(Transform):
+    """Transform that lets you render markdown to HTML in a report."""
+    type = TypeProperty('markdown')
+
+    def get_transform_function(self):
+        def _markdown_text(value):
+            if isinstance(value, str):
+                return mark_safe(markdown.markdown(value))
```

Comment:
is this safe to use as-is?

Reply:
Good point. Wherever we use `mark_safe` now, Bandit automation will give a warning in the future. If you must use `mark_safe`, please also put a comment: `#nosec: reason` for why `mark_safe` was required. In this case, I think we probably want to use [django.utils.html.conditional_escape](https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.html.conditional_escape) on `value`, before it gets passed to `markdown.markdown`. That way, unsafe values will be escaped, but we still give the programmer control to explicitly pass HTML to this.

Reply:
Nice, thanks - that makes a lot of sense. Updated (and added a test case) in https://github.com/dimagi/commcare-hq/pull/31282/commits/6028d18c90d5c17eea366a15e72c6323b9710b51